drivers/dma/dma_mcux_lpc.c

```diff
@@ -89,25 +93,189 @@ static void dma_mcux_lpc_irq_handler(const struct device *dev)
 #endif
 }
 
+static int dma_mcux_lpc_queue_descriptors(struct channel_data *data,
+					   struct dma_block_config *block,
+					   uint8_t src_inc,
+					   uint8_t dest_inc)
+{
+	uint32_t xfer_config = 0U;
+	dma_descriptor_t *next_descriptor = NULL;
+	uint32_t width = data->width;
+	uint32_t max_xfer = NXP_LPC_DMA_MAX_XFER * width;
+	bool setup_extra_descriptor = false;
+	uint8_t enable_interrupt;
+	uint8_t reload;
+	struct dma_block_config local_block;
+
+	memcpy(&local_block, block, sizeof(struct dma_block_config));
+
+	do {
+		/* Descriptors are queued during dma_configure, do not add more
+		 * during dma_reload.
+		 */
+		if (!data->descriptors_queued) {
+			/* Increase the number of descriptors queued */
+			data->num_of_descriptors++;
+
+			/* Do we need to queue additional DMA descriptors for this block */
+			if ((local_block.block_size / width > NXP_LPC_DMA_MAX_XFER) ||
+			    (local_block.next_block != NULL)) {
+				/* Allocate DMA descriptors */
+				next_descriptor = k_aligned_alloc(
```

Comment:
Please use a descriptor pool sizable by he application for this rather than the heap

Reply:
Any negatives to allocating the descriptor pool. Why would we prefer using a Kconfig?

Reply:
I have updated to use `K_HEAP_DEFINE` and also provided a `Kconfig` for the application to adjust the buffer size if needed.

Reply:
Ah I still think this needs to be more like an object pool

See our coding guidelines, which follow misra rules. Rule 4.12 in particular I think applies here.

Reply:
I have updated the code to use an array of DMA descriptors.

Reply:
@teburd, do you think this can go in before this Fridays release freeze?

Reply:
Absolutely, tested this and it fails on intel adsp currently, but its an easy fix. See my comment with a patch. Please apply it and I will +1. Ding me on discord if I don't see it right away.